kafka-connect-aws-s3/src/test/scala/io/lenses/streamreactor/connect/aws/s3/sink/config/S3SinkConfigDefBuilderTest.scala

```diff
@@ -54,14 +54,24 @@ class S3SinkConfigDefBuilderTest extends AnyFlatSpec with MockitoSugar with Matc
 
     element.getStoredAs should be("CSV")
     element.getWithFlushCount should be(1)
-    element.getWithPartitioner should be("Values")
     element.getPartitionBy.asScala.toSet should be(Set("_key"))
+  }
+
+  "S3SinkConfigDefBuilder" should "raises an exception if WITHPARTITIONER is present" in {
+    val props = Map(
+      "connect.s3.kcql" -> s"insert into mybucket:myprefix select * from $TopicName PARTITIONBY _key STOREAS CSV WITHPARTITIONER=Values WITH_FLUSH_COUNT = 1",
+    )
+
+    val ex = CloudSinkBucketOptions(connectorTaskId, S3SinkConfigDefBuilder(props)).left.getOrElse(
+      fail("Expected an exception"),
+    )
 
+    ex.getMessage shouldBe CloudSinkBucketOptions.WithPartitionerError
```

Comment:
EitherValues again:

```suggestion
    CloudSinkBucketOptions(connectorTaskId, S3SinkConfigDefBuilder(props)).left.value.getMessage shouldBe CloudSinkBucketOptions.WithPartitionerError
```

This will automatically throw an exception if it is not a left.

Reply:
Yes. and the test would fail. I don't fully understand the benefit of EitherValue other than a nicer error on the failed test and a deeper stack error.

Reply:
1. Reduce verbosity.
2. Avoid[ logic in unit tests.](https://www.everydayunittesting.com/2016/08/unit-test-anti-patterns-logic-in-tests.html)